src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8673,12 +8673,13 @@ namespace WaterThermalTanks {
         PartLoadRatio = 0.0;
         WaterThermalTank(WaterThermalTankNum).SourceMassFlowRate = 0.0;
         // reset tank inlet temp from previous time step
-        WaterThermalTank(WaterThermalTankNum).SourceInletTemp = WaterThermalTank(WaterThermalTankNum).SourceOutletTemp;
+        WaterThermalTank(WaterThermalTankNum).SourceInletTemp = WaterThermalTank(WaterThermalTankNum).SavedSourceOutletTemp;
         WaterHeaterDesuperheater(DesuperheaterNum).DesuperheaterPLR = 0.0;
 
         Node(WaterInletNode).MassFlowRate = 0.0;
+        Node(WaterInletNode).Temp = WaterThermalTank(WaterThermalTankNum).SavedSourceOutletTemp;
         Node(WaterOutletNode).MassFlowRate = 0.0;
-        Node(WaterOutletNode).Temp = WaterThermalTank(WaterThermalTankNum).SourceOutletTemp;
+        Node(WaterOutletNode).Temp = WaterThermalTank(WaterThermalTankNum).SavedSourceOutletTemp;
```

Comment:
Two things changed here which makes me wonder which one corrected the problem or were both necessary. In any event these changes look reasonable and correct.

Reply:
After fixing the save as what we talked about in the email, the water system energy use changed from 100+ to 40+, those points when desuperheater is "off" are working on well. But for those points when source side is "on", E+ is till giving extreme values. Then I found the ``Tank.SourceInletTemp`` still being calculated  based on the results from previous iteration(not the saved conditions) caused this problem, like the source outlet temperature (``node(waterinletnode).temp``) is set the previous iterated tank temperature which is under extreme condition. So basically the most important change that fixed the problem is "Node(WaterInletNode).Temp = WaterThermalTank(WaterThermalTankNum).SavedSourceOutletTemp;" . Others are set just for consistency.

Reply:
These 2 lines are the only thing about these changes that are of concern. Lines 8676 and 8681. I would think that the water temperature entering the desuperheater is a result of what the tank did this iteration. So it would seem the Tank.SourceOutletTemp would be more accurate. It's important that the tank be simulated based on the final tank temperature from the previous iteration, but the desuperheater, I would think, should use the result from this time step. I know you told me the results looked more accurate after this change but I still have a concern that the desuperheater is not using the correct inlet water temp.

Reply:
It wasn't these initialization lines, it was line 8770 below. When I changed that to use SourceOutletTemp, the water heater Water Heater Source Side Heat Transfer Energy did not match the coil's Water Heater Heating Energy report. So this does look like the correct fix for this issue. I can sleep better now.

Reply:
The water temperature entering the desuperheater is used for calculating the source side temperature entering the tank. Then it calls the ``CalcWaterThermalTankMixed`` function. So I think it should be based on the saved tank conditions from previous time step (if use the previous iteration tank condition, the source inlet temperature would still be crazy and not consistent with tank calculation which is using saved conditions). The lines you pointed out is the outlet node temperature from desuperheater (which enters tank), they would be updated in the following part of codes to make use of reclaimed heat, so it's just an initialization in this place.

Reply:
@rraustad Ok I got it, yes this line is the most important one that fixed the issue. Thanks very much for your help!! Sorry for bringing additional worries 😝

Reply:
Documenting what I found just now. Using this branch the desuperheater coil tracks water tank source energy transfer better (much better) than using the current iteration tank outlet temp as the desuperheater coil inlet temp. So what is done in this branch is very close.

![image](https://user-images.githubusercontent.com/5778156/60131252-ecfd6a80-9766-11e9-87fd-e5c9f87a436f.png)